tls/s2n_connection.c

```diff
@@ -174,6 +175,10 @@ int s2n_connection_wipe(struct s2n_connection *conn)
     GUARD(s2n_stuffer_wipe(&conn->in));
     GUARD(s2n_stuffer_wipe(&conn->out));
 
+    /* Restore the socket option values */
+    GUARD(s2n_socket_read_restore(conn));
+    GUARD(s2n_socket_write_restore(conn));
```

Comment:
s2n_connection_wipe is a sub routine of s2n_connection_new. So we'll do an unnecessary {read,write} socket restore using an uninitialized value when we call s2n_connection_new. Maybe we could add a couple bits of state for "{read,write}_fd_tainted' ? We could use that to track if a restore is required in s2n_connection_wipe.